eng/common/scripts/Verify-RestApiSpecLocation.ps1

```diff
@@ -0,0 +1,121 @@
+function Verify-TspLocation([System.Object]$tspLocationObj) {
+  $repo = $tspLocationObj["repo"]
+  $commit = $tspLocationObj["commit"]
+  if($repo -ne "Azure/azure-rest-api-specs") {
+    LogError "Invalid repo setting in the tsp-location.yaml: $repo. Repo should be 'Azure/azure-rest-api-specs'."
+    exit 1
+  }
+
+  # check the commit hash belongs to main branch
+  Verify-CommitFromMainBranch $commit
+}
+
+# This function is used to verify the specific 'commit' belongs to the main branch of Azure/azure-rest-api-specs repository
+function Verify-CommitFromMainBranch([string]$commit) {
+  try {
+    $searchResult = Search-GitHubCommit -AuthToken $GitHubPat -CommitHash "0f39a2d56070d2bc4251494525cb8af88583a938" -RepoOwner "Azure" -RepoName "azure-rest-api-specs"
+    if ($searchResult.total_count -lt 1) {
+      LogError "Commit $commit doesn't exist in 'main' branch of Azure/azure-rest-api-specs repository."
+      exit 1
+    }
+    else{
+      LogDebug "Commit $commit exists in 'main' branch of Azure/azure-rest-api-specs repository."
+    }
+  }
+  catch {
+    LogError "Failed to search commit $commit with exception:`n$_"
+    exit 1
+  }
+}
+
+try{
+  $ServiceDir = Join-Path $RepoRoot 'sdk' $ServiceDirectory
+  $PackageDirectory = Join-Path $ServiceDir $PackageName
+  Push-Location $PackageDirectory
+
+  # Load tsp-location.yaml if existed
+  $tspLocationYamlPath = Join-Path $PackageDirectory "tsp-location.yaml"
+  $autorestMdPath = Join-Path $PackageDirectory "src/autorest.md"
+  $tspLocationYaml = @{}
+  if (Test-Path -Path $tspLocationYamlPath) {
+    # typespec scenario
+    $tspLocationYaml = Get-Content -Path $tspLocationYamlPath -Raw | ConvertFrom-Yaml
+    Verify-TspLocation $tspLocationYaml
+  }
+  elseif ($Language -eq "dotnet") {
+    # only dotnet language sdk uses 'autorest.md' to configure the sdk generation, ignore this validation for the other languages
+    if (Test-Path -Path $autorestMdPath) {
+      $autorestMdContent = Get-Content -Path $autorestMdPath -Raw
+      $yamlContent = $autorestMdContent -split '``` yaml|```'
+      $yamlSection = $yamlContent[1]
+      if ($yamlSection) {
+        $yamlobj = ConvertFrom-Yaml -Yaml $yamlSection
+        $requireValue = $yamlobj["require"]
+        $inputFileValue = $yamlobj["input-file"]
+        if ($requireValue) {
+          LogDebug "require is set as:$requireValue"
+          Verify-AutorestMd $requireValue
+        }
+        elseif ($inputFileValue) {
+          LogDebug "input-file is set as:$inputFileValue"
+          foreach($inputFile in $inputFileValue) {
+            Verify-AutorestMd $inputFile
+          }
+        }
+        else {
+          LogWarning "Both 'require' and 'input-file' haven't been found in $autorestMdPath."
+          exit 0
+        }
+      }
+    }
+    else {
+      LogWarning "autorest.md hasn't been found in $PackageDirectory for language: $Language"
```

Comment:
How do we not show these warnings for any project that isn't generated? We need to be careful to not start causing a bunch of noise in the builds.

Reply:
Removed.